lib/rpmsg/rpmsg_virtio.c

```diff
@@ -410,22 +410,21 @@ static void rpmsg_virtio_rx_callback(struct virtqueue *vq)
 		ept = rpmsg_get_ept_from_addr(rdev, rp_hdr->dst);
 		metal_mutex_release(&rdev->lock);
 
-		if (!ept)
-			/* Fatal error no endpoint for the given dst addr. */
-			return;
+		if (ept) {
+			if (ept->dest_addr == RPMSG_ADDR_ANY) {
+				/*
+				 * First message received from the remote side,
+				 * update channel destination address
+				 */
+				ept->dest_addr = rp_hdr->src;
+			}
+			status = ept->cb(ept, RPMSG_LOCATE_DATA(rp_hdr),
+					 rp_hdr->len, rp_hdr->src, ept->priv);
 
-		if (ept->dest_addr == RPMSG_ADDR_ANY) {
-			/*
-			 * First message received from the remote side,
-			 * update channel destination address
-			 */
-			ept->dest_addr = rp_hdr->src;
+			RPMSG_ASSERT(status == RPMSG_SUCCESS,
+				     "unexpected callback status\n");
 		}
```

Comment:
better to have RPMSG_ASSERT for  for if (!ept) case.

Reply:
The message come from outside world, should we more robust here to avoid the DoS attack?
Actually, Linux kernel do the similar thing in rpmsg_recv_single:
https://github.com/torvalds/linux/blob/master/drivers/rpmsg/virtio_rpmsg_bus.c

Reply:
agreed.